session.go

```diff
@@ -474,8 +477,10 @@ func (s *session) retry(goCtx goctx.Context, maxCnt int) error {
 			s.sessionVars.StmtCtx.ResetForRetry()
 			_, err = st.Exec(goCtx)
 			if err != nil {
+				s.Txn().StmtRollback()
 				break
 			}
+			terror.Log(s.Txn().StmtCommit())
```

Comment:
I think we need to rollback the whole transaciton when StmtCommit return error. Because it may already make the buffer dirty.

Reply:
StmtCommit should never meet error...  (some thing similar to malloc() return error)
Handle those rare case is not worthy

Reply:
Ok, I hope so.